Task #1607: Apply New Styles to Package Documentation Module Navigation

henilGondalia opened the pull request:
Resolves #1607

This PR addresses issue #1607, focusing on applying new style to Package Documentation Module Navigation of the OCaml.org website. [Here's a Figma reference design](https://www.figma.com/file/J0w4b9exaHhVpvabKoGqTn/Design-Github-Issues?type=design&node-id=27%3A136&mode=design&t=KUzqK4w2r2nJuXGF-1). 

## Screenshots of New Style

![Doc Ocaml 1](https://github.com/ocaml/ocaml.org/assets/34880491/4ffd4e33-26f1-4a6a-84ba-e05dff05f5df)

![Doc Ocaml 3](https://github.com/ocaml/ocaml.org/assets/34880491/11c7306e-3d6f-45f6-9a9c-9a52a98d4e62)


Please review the changes and let me know if any further adjustments are needed. Thank you!

henilG committed:
Apply New Styles to Package Documentation Module Navigation

sabine commented:
Hey that looks really beautiful!

Here's some points to watch out for:

1. scroll bar overlapping +/- symbols
![Screenshot from 2023-10-16 10-25-22](https://github.com/ocaml/ocaml.org/assets/6594573/55367c21-abe1-40c7-9ba1-89153fa87f7a)

2. The three parameters here are at the same level in the hierarchy, but their left margin is different (see https://ocaml.org/p/irmin/3.8.0/doc/Irmin/Storage/Append_only/argument-2-K/index.html):
![Screenshot from 2023-10-16 10-27-49](https://github.com/ocaml/ocaml.org/assets/6594573/80a0823f-393f-4a8f-a50d-164d35c8e193)

3. Click area for +/- symbols is very small (I recommend adding horizontal padding):
![Screenshot from 2023-10-16 10-30-07](https://github.com/ocaml/ocaml.org/assets/6594573/a128218c-29ef-4d10-bcbe-95b4380bbc63)

henilGondalia commented:
Thank you, @sabine for appreciation. I'll work on implementing the suggested changes and fixing bugs.

henilG committed:
Fixes Scrollbar overlapping, margin issue, and added horizonal padding to +/- symbol for better user experience.

henilGondalia commented:
Hello @sabine as you suggested I have added some horizonal space for +/- and also to give some space from the scrollbar. Also fixed space issue in same level in the hierarchy.
Please review these updates and let me know your feedback.

sabine commented:
I'm impressed, this looks really great to me @henilGondalia! Thanks for the contribution!

I'm putting this on staging.ocaml.org for others to test as well.

sabine commented on asset/css/doc.css:
```diff
@@ -34,7 +34,7 @@ div.odoc .spec a.anchor {
   left: 0;
   top:0.3em;
-  opacity: 0;
+  opacity: 1;
```
this change seems unrelated to the package documentation navigation

sabine commented on asset/css/doc.css:
```diff
@@ -47,19 +47,24 @@ div.odoc .spec a.anchor::after {
+} */
+
+div.odoc *>a.anchor {
+  color: rgb(75 85 99);
```
seems unrelated to the package navigation documentation navigation

henilGondalia replied:
Initially, I thought it was related to the hover CSS of navigation line, so I made changes to it. However, upon checking again, I realized that it wasn't that css, as you rightly pointed out. It is a mistake which I will revert with lib indicator css issue.

sabine commented on asset/css/doc.css:
```diff
@@ -201,9 +206,10 @@ div.odoc .comment-delim {
+/* span.icon-expand:not(.open)>.navmap-tag,
+span.no-expand>.navmap-tag {
+  background: white;
+} */
```
remove, instead of commenting

sabine commented on asset/css/doc.css:
```diff
@@ -273,30 +285,43 @@ span.icon-expand:not(.open) > .navmap-tag , span.no-expand > .navmap-tag {
-span.no-expand > .navmap-tag {
-  margin-left: 12px;
+span.icon-expand>.navmap-tag,
+span.no-expand>.navmap-tag {
```
```suggestion
span.icon-expand > .navmap-tag,
span.no-expand > .navmap-tag {
```

for better readability

sabine reviewed (commented):
I noticed one last very little thing:

![Screenshot from 2023-10-17 13-25-35](https://github.com/ocaml/ocaml.org/assets/6594573/b701eb25-7495-474b-9564-55d2811e6fb2)

At desktop screen size, the little indicator line left of "lib" is cut off a bit and thus looks thinner than the others.

henilG committed:
Fix Lib indicator cut-off, remove unnecessary code, reverted unrelated code back

henilGondalia commented:
Hello @sabine,
Is there any other task that I can take on ?

sabine commented:
@henilGondalia there's more applicants than issues still. I have ideas for two more medium issues I can set up tomorrow, but should probably assign folks that didn't have a chance to try a medium issue yet.

sabine commented on asset/css/doc.css:
```diff
@@ -47,7 +47,7 @@ div.odoc .spec a.anchor::after {
 }
 
-div.odoc *:hover > a.anchor {
+ div.odoc *:hover > a.anchor {
```
```suggestion
div.odoc *:hover > a.anchor {
```

sabine reviewed (approved):
Thanks for the contribution @henilGondalia! Excellent work!

sabine committed:
Update asset/css/doc.css

tmattio commented:
I absolutely love the new package nav UI! Thank you for your work @henilGondalia!

henilGondalia commented:
> @henilGondalia there's more applicants than issues still. I have ideas for two more medium issues I can set up tomorrow, but should probably assign folks that didn't have a chance to try a medium issue yet.

Sure, if there's no one else available, I'd be happy to take on the task.

> Thanks for the contribution @henilGondalia! Excellent work!

Thank you @sabine for appreciation. It was a pleasure to contribute! 😊

henilGondalia commented:
> I absolutely love the new package nav UI! Thank you for your work @henilGondalia!

Thank you @tmattio. I'm glad that you love it.

shakthimaan commented:
Thank you for your valuable feedback! I am closing this PR in favour of https://github.com/ocaml/ocaml.org/pull/1854 to server the manuals directly from ocaml.org.

Octachron commented:
Since I am not sure how long switching to `ocaml/ocaml.org` will take, I will open a PR for patching the manual for 5.1.1 on `v2.ocaml.org` while waiting for the update to avoid propagating incorrect information.
